[Android] Update README.md (#26)
sendbird/sendbird-ai-agent: android/README.md

[iOS](https://github.com/sendbird/sendbird-ai-agent/blob/main/ios/README.md) / **Android** / [JS](https://github.com/sendbird/sendbird-ai-agent/blob/main/js/README.md)

# Sendbird AI Agent Quickstart guide (Android)

The **Sendbird AI Agent Messenger** allows seamless integration of chatbot features into your Android application. Follow the steps below to initialize and utilize the SDK effectively.

- [Sendbird AI Agent Quickstart guide (Android)](#sendbird-ai-agent-quickstart-guide-android)
  - [Requirements](#requirements)
  - [Prerequisites](#prerequisites)
  - [Getting started](#getting-started)
    - [Step 1. Create a new project](#step-1-create-a-new-project)
    - [Step 2. Install AI Agent SDK](#step-2-install-ai-agent-sdk)
    - [Step 3. Initialize AI Agent SDK](#step-3-initialize-ai-agent-sdk)
  - [Running your application](#running-your-application)

>     - [Manage sessions](#manage-sessions)

Author: ```suggestion
    - [Manage user sessions](#manage-user-sessions)
```

- Android Gradle plugin 8.0.0 or higher
- Android ViewBinding enabled

## Prerequisites

Before you start, you'll need your Sendbird **Application ID** and **AI Agent ID**. 
<br><br/>
You can find it under the **Channels** > **Messenger** menu on the Sendbird Dashboard.

![ai-agent-app-id-agent-id](https://github.com/user-attachments/assets/37d2873e-f35d-45dd-97cc-3d7c7e638a0c)

---

## Getting started

Quickly install and initalize the AI Agent SDK by following the steps below.

### Step 1. Create a new project

1. In Android Studio, create a new project (**File > New > New Project...**).

> 2. Select **Empty Activity** and click **Next**.

Reviewer: ```suggestion
2. Select **Empty Views Activity** and click **Next**.
```
@min2jane 
Empty Views Activity 로 변경되는게 좋아보이고, 이미지도 클릭되는 부분이 변경되면 더 좋아보입니다!
Empty Activity는 Compose인데 아직 지원하지 않고있어서요

![ai-agent-android-new-project](https://github.com/user-attachments/assets/3507e758-476a-4118-a3ae-9980103f83f1)

> 3. Give your project a name. Accept all other defauls, including the langauge as `Kotlin` and the minimum SDK as `API 21: Android 5.0 (Lollipop)` and click **Finish**.

Author: ```suggestion
3. Give your project a name. Accept all other defaults, including the language as `Kotlin` and the minimum SDK as `API 21: Android 5.0 (Lollipop)` and click **Finish**.
```

> ![ai-agent-android-new-project-settings](https://github.com/user-attachments/assets/9b9808fa-70c5-4a52-ab77-2d2c1638cc23)

Reviewer: @min2jane 요거 이미지도 Compose용 Empty Activity네요

Reviewer: 이미지 캡쳐 새로 해야한다면 바꾸는 김에... Project name도 바꾸는것도!!

           }
       }
       ```

>        **Note:** You should be using Gradle 6.8 or higher. You can check the `gradle-wrapper.properties` file in your project to see the version of Gradle you are using.

Reviewer: ```suggestion
       **Note:** You should be using Gradle 8.0 or higher. You can check the `gradle-wrapper.properties` file in your project to see the version of Gradle you are using.
```

    }
    ```

#### How to create the Application file

1. Open Android Studio and navigate to your project.
2. In the left side bar, locate the **app** module.
3. Right-click on the java or kotlin folder (depending on the language of your project).
4. Select **New > Kotlin File/Class**.
5. In the dialog that appears, select **Class** and anme it `Application`.
6. Then, add the code above to your `Application.kt` file.
   
---

## Running your application

Now that you have installed and initialized the AI Agent SDK, follow the steps below to run your application.

> ### Manage sessions

Author: ```suggestion
### Manage user sessions
```

    
To use the SDK, session information is required.  
Before using `AIAgentMessenger`, you must set the session information for the application user issued by Sendbird.

#### When to set the session information:
  
1. **Upon initial login:**  
Since there is no session information for the user at the first login, it should be set immediately after app login.

2. **For already logged-in users:**  
If the user is already logged in, the session should be registered immediately after SDK initialization.

        ```kotlin
        val userSessionInfo = UserSessionInfo(userId, authToken, AbstractSessionHandler())
        AIAgentMessenger.updateSessionInfo(userSessionInfo)
        ```
     
### Handle session expiration

When registering a session information, the provided session handler must handle the case where the session information expires.  
Follow the code below to refresh and provide a new session token in the session handler callback.

#### How to handle session expiration:
     
1. Detect session expiration in the session handler callback.
2. Request a new session token from the server.
3. Update the SDK with the new session token.
   
        ```kotlin

>         class AbstractSessionHandler : AIAgentSessionHandler() {

Reviewer: @leo-shin 요거는 Messenger 용으로 따로 빼지는 않고 AI Agent면 모두 이용할 수 있도록 하는거겠죠?

        }
        ```

### Launch the messenger

Once the authentication information has been successfully registered, you can launch the messenger to start a conversation with the ai agent.  

> There are two ways to do this:

Author: ```suggestion
There are two ways to display the messenger:
```

> 1. Using the messenger launcher button

Author: ```suggestion
1. Using the launcher button
```

> 1. Using the launcher button

Reviewer: @min2jane 안드로이드에서 Launcher button을 사용 할 때 또한 2가지 방법으로 열 수 있도록 기능이 구현되어있는데요,
인자를 넘겨주지 않거나 (Default) FullScreen로 넘겨준다면 Full Screen Dialog 모드로 뜨게 되고, 
'messengerLauncher.init(it)' or 'messengerLauncher.init(it, DialogConfig.FullScreen)'

Anchor로 넘겨준다면
messengerLauncher.init(it, DialogConfig.Anchored(binding.aiAgentLauncher, DialogDirection.DEFAULT))
Launcher를 기준으로 삼아서 Launcher위에 Dialog가 보여지는 기능이 있는데요,
이 부분을 여기에 추가하는게 좋을까요?
아니면 맨 아래에 추가 기능에 넣는게 좋을까요?

Reviewer: 참고로 아래에 FullScreen과 FullScreenDialog는 다른 모드입니다 👍 
FullScreen은 진짜로 창 전체를 다 덮는 것이고, FullScreenDialog는 팝업 창이 화면 전체를 덮고, 아주 조금의 여백이 있어서 뒤에 화면이 조금 보이는 상태라고 이해하시면 편합니다.

Author: @elliot-choic 
그렇군요! 들어간다면 이 Using the launcher button 에 a ,b 정도로 나눠서 넣어놓는게 좋을것 같은데요,
우선 이 부분이 **퀵스타트 가이드**에 설명이 꼭 들어가야 하는 부분인지 궁금합니다! 
(추후에 조금더 상세하게 다룰수 있는 부분이 있다면 글로 빠져서 들어가는게 좋을것 같기도 해서 제안드립니다용~)

Reviewer: @min2jane 음 🤔 그럼 빠지는것도 좋아보이네요
Custom launcher button을 사용하는 가이드도 추가로 써야할것 같긴한데 그때 적는것도 좋아보이네요
그러면 다른 모드가 있다라는걸 인지해주는 정도로 코드 스니펫을
messengerLauncher.init(it) -> messengerLauncher.init(it, DialogConfig.FullScreen)
요정도로 변경해놓을까요?

Reviewer: 취소요!!! 지금 그대로 놓는게 좋겠네요! 추후 다른 곳에서 다같이 적는게 좋을것 같습니다 ㅋㅋ

        messengerLauncher.init(aiAgentId = "your_ai_agent_id")
        ```

#### 2. Opening the conversation channel in full-screen mode

![Image](https://github.com/user-attachments/assets/348ccad1-ec9a-4851-9324-084eaf569e34)
    
You can open a full-screen conversation by starting an `Activity`.  
    
    ```kotlin
    startActivity(ConversationActivity.newIntent(context, "your_ai_agent_id"))
    ```

---

## Advanced features

> THe following are available advanced features.

Author: ```suggestion
The following are available advanced features.
```
